cpp/calculate-demand.cpp

```diff
@@ -162,17 +183,17 @@ long calculateUnsatisfiedDemand(std::string targetFilename, std::string demoFile
 #endif
 
     // Second pass: we apply the multiplying factor to each pixel
-    for (int iXBlock = 0; iXBlock < nXBlocks; ++iXBlock) {
+    for (int iXBlock = 0; iXBlock < facilityNXBlocks; ++iXBlock) {
       xOffset = iXBlock*xBlockSize;
       nXValid = xBlockSize;
-      if (iXBlock == nXBlocks-1) nXValid = xSize - xOffset;
+      if (iXBlock == facilityNXBlocks-1) nXValid = facilityXSize - xOffset;
```

Comment:
Good catch. Yes, I think it will be good to add that check but outside the loop so we check it just once and the error, if it occurs, it does in a more reasonable place. Something like: `assert(facilitySize < (demoSize - (blockSize * blocksOffset))`